### PR TITLE
Auto correct the IP if the operator set their IP to localhost when opt-in

### PR DESCRIPTION
**Pull request by wmb-software-consulting**
## Why are these changes needed?

We are seeing more operators setting their Node IP to localhost.
Such Nodes won't be reachable from public. We should auto correct the IP (from localhost to current IP with IP checker) when the operator registers.

## Checks

- [x] I've made sure the lint is passing in this PR.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, in that case, please comment that they are not relevant.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Integration tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: node/plugin/config.go [hunk 1]
```diff
@@ -14,6 +14,31 @@ import (
 var (
 	/* Required Flags */
 
+	HostnameFlag = cli.StringFlag{
```
Comment (jianoaix) [inline node/plugin/config.go:17]: All these 3 flags are already encoded in the existing `SocketFlag`, which you may parse. So there is no need to redefine them here again.

**Comment by jianoaix** [comment]
@shrimalmadhur This can affect the operator setup or experience.

File: node/plugin/cmd/main.go [hunk 2]
```diff
@@ -114,8 +116,33 @@ func pluginOps(ctx *cli.Context) {
 		return
 	}
 
+	s := strings.Split(config.Socket, ":")
```
Comment (jianoaix) [inline node/plugin/cmd/main.go:119]: Can we make this a method (like ParseOperatorSocket) next to https://github.com/Layr-Labs/eigenda/blob/66c1e06c51c7aedb951ede024627e42af21a7a3b/core/state.go#L17?

File: core/state.go [hunk 3]
```diff
@@ -18,6 +19,26 @@ func MakeOperatorSocket(nodeIP, dispersalPort, retrievalPort string) OperatorSoc
 	return OperatorSocket(fmt.Sprintf("%s:%s;%s", nodeIP, dispersalPort, retrievalPort))
 }
 
+func ParseOperatorSocket(operatorSocket string) (socket string, dispersalPort string, retrievalPort string, err error) {
```
Comment (jianoaix) [inline core/state.go:22]: The first return value should be host, right?

File: node/plugin/config.go [hunk 4]
```diff
@@ -14,6 +14,13 @@ import (
 var (
 	/* Required Flags */
 
+	PubIPProviderFlag = cli.StringFlag{
+		Name:     "public-ip-provider",
+		Usage:    "The ip provider service used to obtain a operator's public IP [seeip (default), ipify)",
+		Required: true,
```
Comment (jianoaix) [inline node/plugin/config.go:20]: Does this need to be required? @shrimalmadhur

File: core/state.go (hunk 3 above)
Comment (wmb-software-consulting) [reply core/state.go:22]: Ok, I gotcha what you mean.

File: core/state.go [hunk 5]
```diff
@@ -18,6 +19,25 @@ func MakeOperatorSocket(nodeIP, dispersalPort, retrievalPort string) OperatorSoc
...
 }
 
+func ParseOperatorSocket(operatorSocket string) (host string, dispersalPort string, retrievalPort string, err error) {
+	s := strings.Split(operatorSocket, ":")
+	if len(s) != 2 {
+		err = fmt.Errorf("invalid socket address format: %s", operatorSocket)
+		return
+	}
+	host = s[0]
+	dispersalPort = s[1]
```
Comment (jianoaix) [inline core/state.go:29]: Is this correct? If "host:123;456" is split by ":", then the s[1] will be "123;456", right?

Comment (wmb-software-consulting) [reply core/state.go]: Nice catch!

File: core/state.go [hunk 6]
```diff
@@ -18,6 +19,25 @@ func MakeOperatorSocket(nodeIP, dispersalPort, retrievalPort string) OperatorSoc
...
+
+	s = strings.Split(s[0], ":")
+	if len(s) != 2 {
+		err = fmt.Errorf("invalid socket address format: %s", operatorSocket)
+		return
+	}
+	host = s[0]
+	dispersalPort = s[1]
+
+	return
```
Comment (jianoaix) [inline core/state.go:38]: nit: this return isn't needed

File: core/test/core_test.go [hunk 7]
```diff
@@ -256,3 +256,12 @@ func TestCoreLibrary(t *testing.T) {
 	}
 
 }
+
+func TestParseOperatorSocket(t *testing.T) {
```
Comment (jianoaix) [inline core/test/core_test.go:260]: Thanks for adding this test. Maybe add a couple invalid cases.

File: node/plugin/cmd/main.go [hunk 8]
```diff
@@ -97,8 +99,24 @@ func pluginOps(ctx *cli.Context) {
 		return
 	}
 
+	host, dispersalPort, retrievalPort, err := core.ParseOperatorSocket(config.Socket)
+	if err != nil {
+		log.Printf("Error: failed to parse operator socket: %v", err)
+		return
+	}
+
+	socket := string(core.MakeOperatorSocket(host, dispersalPort, retrievalPort))
```
Comment (jianoaix) [inline node/plugin/cmd/main.go:108]: It looks you are doing some back and forth (parsing and then making again). Can it just use `config.Socket`?

**Review by jianoaix — COMMENTED** [review]
LGTM

File: core/state.go (hunk 6 above)
Comment (wmb-software-consulting) [reply core/state.go:38]: Actually, it's needed so

**Review by jianoaix — APPROVED** [review]

File: node/plugin/config.go (hunk 4 above)
Comment (shrimalmadhur) [reply node/plugin/config.go:20]: oh umm - I don't think we should put it as required - can we make this as optional and by default point to `seeip`? 

We can re-use the `.env` one in the setup

**Pull request merged**